hail/python/hailtop/aiocloud/aioazure/fs.py

```diff
@@ -156,7 +157,7 @@ async def read(self, n: int = -1) -> bytes:
 
         if n == -1:
             try:
-                downloader = await self._client.download_blob(offset=self._offset)
+                downloader = await self._client.download_blob(offset=self._offset, length=self._length)
```

Comment:
I can't remember exactly, but I might have run into issues with Azure when the file is empty and you ask for 0 bytes.

Reply:
Totally right. I can add a helpful error message for azure.

```
HttpResponseError: The range specified is invalid for the current size of the resource.
RequestId:314c0805-301e-001f-64f6-547b9b000000
Time:2022-04-20T20:38:34.1301964Z
ErrorCode:InvalidRange
```

Reply:
Hrm. that's annoying to have one give an error, so I'll just have it return an empty readable.